File: app/src/main/java/kr/com/misemung/realm/repository/CityRepository.java
package kr.com.misemung.realm.repository;



import io.realm.Realm;
import io.realm.RealmResults;
import kr.com.misemung.realm.entity.AirRecord;
import kr.com.misemung.realm.entity.CityRecord;
import kr.com.misemung.vo.AirInfo;
import kr.com.misemung.vo.CityInfo;

public class CityRepository {

    public static class City {
        public static void set(String umdName, CityInfo cityInfo) {
            Realm realm = Realm.getDefaultInstance();

            try {
                realm.beginTransaction();

                Number nextID = (realm.where(AirRecord.class).max("id"));
                if (nextID == null) {
                    nextID = 1;
                } else {
                    nextID = nextID.intValue() + 1;
                }

                CityRecord record = new CityRecord(cityInfo);
                record.umdName = umdName;
                record.id = (int) nextID;

                realm.insertOrUpdate(record);
                realm.commitTransaction();
            } catch (Exception e) {
                e.printStackTrace();
                realm.cancelTransaction();
            } finally {
                realm.close();
            }
        }

        public static RealmResults<CityRecord> selectByCityList() {

            return Realm.getDefaultInstance().where(CityRecord.class)
                    .findAll();
        }

        public static CityRecord selectByCityData(int id) {

            return Realm.getDefaultInstance().where(CityRecord.class)
                    .equalTo("id", id)
                    .findFirst();
        }

        public static void deleteCityData(int id) {
            Realm realm = Realm.getDefaultInstance();

            realm.executeTransaction(realm1 -> {
                RealmResults<CityRecord> result
                        = realm1.where(CityRecord.class).equalTo("id", id).findAll();
                result.deleteAllFromRealm();
            });
        }

    }
}

File: app/src/main/java/kr/com/misemung/realm/repository/CityRepository.kt
package kr.com.misemung.realm.repository

import io.realm.Realm
import kr.com.misemung.realm.entity.AirRecord
import kr.com.misemung.realm.entity.CityRecord
import kr.com.misemung.vo.CityInfo

class CityRepository {
    object City {
        operator fun set(umdName: String?, cityInfo: CityInfo) {
            val realm = Realm.getDefaultInstance()
            try {
                realm.beginTransaction()
                var nextID = realm.where(AirRecord::class.java).max("id")
                nextID = if (nextID == null) {
                    2
                } else {
                    nextID.toInt() + 1
                }
                val record = CityRecord(cityInfo)
                record.umdName = umdName
                record.id = nextID
                realm.insertOrUpdate(record)
                realm.commitTransaction()
            } catch (e: Exception) {
                e.printStackTrace()
                realm.cancelTransaction()
            } finally {
                realm.close()
            }
        }

        fun setCurrentCity(id: Int, umdName: String?, cityInfo: CityInfo) {
            val realm = Realm.getDefaultInstance()
            try {
                realm.beginTransaction()
                val record = CityRecord(cityInfo)
                record.umdName = umdName
                record.id = id
                realm.insertOrUpdate(record)
                realm.commitTransaction()
            } catch (e: Exception) {
                e.printStackTrace()
                realm.cancelTransaction()
            } finally {
                realm.close()
            }
        }

        fun selectByCityData(id: Int): CityRecord? {
            return Realm.getDefaultInstance().where(CityRecord::class.java)
                .equalTo("id", id)
                .findFirst()
        }

        fun deleteCityData(id: Int) {
            val realm = Realm.getDefaultInstance()
            realm.executeTransaction { realm1: Realm ->
                val result = realm1.where(
                    CityRecord::class.java
                ).equalTo("id", id).findAll()
                result.deleteAllFromRealm()
            }
        }
    }
}